crates/libcgroups/src/v2/devices/emulator.rs

```diff
@@ -54,4 +54,67 @@ impl Emulator {
     }
 }
 
-// FIXME: add some tests
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use oci_spec::runtime::LinuxDeviceCgroupBuilder;
+
+    #[test]
+    fn test_with_default_allow() {
+        // act
+        let emulator = Emulator::with_default_allow(true);
+
+        // assert
+        assert_eq!(emulator.rules.len(), 0);
+        assert_eq!(emulator.default_allow, true);
+    }
+
+    #[test]
+    fn test_type_a_rule() {
+        // arrange
+        let mut emulator = Emulator::with_default_allow(false);
+        let cgroup = LinuxDeviceCgroupBuilder::default()
+            .typ(LinuxDeviceType::A).build().unwrap();
+
+        // act
+        emulator.add_rule(&cgroup).expect("add type A rule");
+
+        // assert
+        assert_eq!(emulator.rules.len(), 0);
+        assert_eq!(emulator.default_allow, false);
+    }
+
+    #[test]
+    fn test_add_empty_rule() {
+        // arrange
+        let mut emulator = Emulator::with_default_allow(false);
+        let cgroup = LinuxDeviceCgroupBuilder::default()
+            .build().unwrap();
+
+        // act
+        emulator.add_rule(&cgroup).expect("add empty rule");
+
+        // assert
+        assert_eq!(emulator.rules.len(), 0);
+        assert_eq!(emulator.default_allow, false);
+    }
+
+    #[test]
+    fn test_add_some_rule() {
+        // arrange
+        let mut emulator = Emulator::with_default_allow(false);
+        const PERMISSION: &str = "PERMISSION";
```

Comment:
Is there a better way? I didn't look too far into what this param is supposed to be, at this point I'm just trying to successfully exercise code paths.